armi/physics/neutronics/latticePhysics/latticePhysicsWriter.py

```diff
@@ -409,8 +409,10 @@ def _adjustPuFissileDensity(self, nucDensities):
         fiss = sum(dens[0] for nuc, dens in nucDensities.items() if nuc.isFissile())
         hm = sum(dens[0] for nuc, dens in nucDensities.items() if nuc.isHeavyMetal())
 
-        if fiss / hm < minFrac:
+        if hm and fiss / hm < minFrac:
             pu239 = nuclideBases.byName["PU239"]
+            if not pu239:
+                return
```

Comment:
Consider updating the `minimumFissileFraction` user input rather than updating the code. TBH, this is a weird code-specific assumption that should probably be removed. I added a warning in that method a long time ago, but haven't gotten around to handling it.

```
            runLog.warning(
                "Adjusting Pu-239 number densities in {} from {} to {} to meet minimum fissile fraction "
                "of {}.\nNote: This modeling approximation will be deprecated soon so it is recommended to "
                "drive this composition with an external source.".format(
                    self.block, old, new, minFrac
                )
            )
```

Reply:
Also, is there a reason by `nuclideBases` would not have a Pu-239 nuclide? It should be there.. I would expect that you want to handle this specifically on the `old, temp, msg = nucDensities[pu239]`  call instead. Could consider updating U235 instead.

Reply:
Actually I think I was fooling myself when I hit this error.